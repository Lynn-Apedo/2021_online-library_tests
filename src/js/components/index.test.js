import Ban from './ban'
import Button from './button';
import { render, screen } from '@testing-library/react'
// import Layout from './layout'
import Home from '../pages/home'
import { BrowserRouter as Router, Redirect, Route, Switch, Link } from "react-router-dom";
import renderer from 'react-test-renderer'
 
describe('Ban', () => {
    it('Should render without crash', async () => {
        render(
            <Router>
                <Ban />
            </Router>
        )
    })
    it('Should render h1 link', async () => {
        render(
            <Router>
                <Ban />
            </Router>
        )
        expect(screen.getByText('EZE')).toBeTruthy()
    })
    it('Should matches snapshot', () => {
        const saveTitleBan = renderer.create(<Router><Ban /></Router>).toJSON()
        expect(saveTitleBan).toMatchSnapshot()
    })
})


// describe('Button', () => {
//     it('Should render button with the right name', async () => {
//         render(
//             <Button />
//         )
//     expect(screen.getByText("M'enregistrer")).toBeTruthy()
//     })
// })

// describe('authRoute', () => {
//     it('Should redirect to /register if isLogged is not true', async () => {
//         const isLogged = false
//         if (isLogged == false) {
//             render(
//                 <Router>
//                     <Redirect to="/register" />
//                 </Router>
//             )
//         }
//     })
// })