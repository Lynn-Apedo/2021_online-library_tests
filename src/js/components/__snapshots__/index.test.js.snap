// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ban Should matches snapshot 1`] = `
<div>
  <a
    href="/"
    onClick={[Function]}
  >
    <h1>
      EZE
    </h1>
  </a>
  <div
    className="ban"
  />
</div>
`;
